pkg/parcacol/sample.go

```diff
@@ -224,100 +225,46 @@ func SeriesToArrowRecord(
 		}
 	}
 
-	return bldr.NewRecord(), nil
-}
-
-// ParquetBufToArrowRecord converts a parquet buffer to an arrow record. If rowsPerRecord is 0, then the entire buffer is converted to a single record.
-func ParquetBufToArrowRecord(ctx context.Context, mem memory.Allocator, buf *dynparquet.Buffer, s *dynparquet.Schema, rowsPerRecord uint) ([]arrow.Record, error) {
-	as, err := pqarrow.ParquetSchemaToArrowSchema(ctx, buf.Schema(), s, logicalplan.IterOptions{})
-	if err != nil {
-		return nil, err
-	}
-
-	bldr := array.NewRecordBuilder(mem, as)
-	defer bldr.Release()
-
-	rows := buf.Rows()
-	defer rows.Close()
-
-	buffSize := 256
-	rowBuf := make([]parquet.Row, buffSize)
-
-	records := []arrow.Record{}
-	recordSize := uint(0)
-	for {
-		n, err := rows.ReadRows(rowBuf)
-		if err != nil && err != io.EOF {
-			return nil, err
+	sortingColumns := []arrowutils.SortingColumn{}
+	arrowFields := as.Fields()
+	for _, col := range schema.SortingColumns() {
+		direction := arrowutils.Ascending
+		if col.Descending() {
+			direction = arrowutils.Descending
 		}
 
-		if n == 0 {
-			break
+		colDef, found := schema.ColumnByName(col.ColumnName())
+		if !found {
+			return nil, fmt.Errorf("sorting column %v not found in schema", col.ColumnName())
 		}
 
-		// Write each row to the arrow record
-		for _, r := range rowBuf[:n] {
-			for i := range r {
-				switch as.Field(i).Name {
-				case profile.ColumnName:
-					fallthrough
-				case profile.ColumnPeriodType:
-					fallthrough
-				case profile.ColumnPeriodUnit:
-					fallthrough
-				case profile.ColumnSampleType:
-					fallthrough
-				case profile.ColumnSampleUnit:
-					if err := bldr.Field(i).(*array.BinaryDictionaryBuilder).AppendString(r[i].String()); err != nil {
-						return nil, err
-					}
-				case profile.ColumnStacktrace:
-					bldr.Field(i).(*array.BinaryBuilder).AppendString(r[i].String())
-				case profile.ColumnDuration:
-					fallthrough
-				case profile.ColumnPeriod:
-					fallthrough
-				case profile.ColumnTimestamp:
-					fallthrough
-				case profile.ColumnValue:
-					bldr.Field(i).(*array.Int64Builder).Append(r[i].Int64())
-				default:
-					switch {
-					case strings.HasPrefix(as.Field(i).Name, profile.ColumnPprofNumLabels):
-						if r[i].IsNull() {
-							bldr.Field(i).AppendNull()
-						} else {
-							bldr.Field(i).(*array.Int64Builder).Append(r[i].Int64())
-						}
-					case strings.HasPrefix(as.Field(i).Name, profile.ColumnPprofLabels):
-						fallthrough
-					case strings.HasPrefix(as.Field(i).Name, profile.ColumnLabels):
-						if r[i].IsNull() {
-							bldr.Field(i).AppendNull()
-						} else {
-							if err := bldr.Field(i).(*array.BinaryDictionaryBuilder).AppendString(r[i].String()); err != nil {
-								return nil, err
-							}
-						}
-					default:
-						panic(fmt.Sprintf("unknown column %v", as.Field(i).Name))
-					}
+		if colDef.Dynamic {
+			for i, c := range arrowFields {
```

Comment:
Kind of annoying to have to iterate over all fields for each dynamic column. What if we created a map with the field prefix so this can instead be an O(1) lookup? Again, not sure if worth optimizing.